### PR TITLE
add chain head to agent info

### PR DESCRIPTION
**Pull request by thedavidmeister**
### Summary

- add current chain head to agent info

### TODO:
- [x] CHANGELOG(s) updated with appropriate info

### CODE REVIEW

File: crates/holochain_zome_types/src/info.rs [hunk 1]
```diff
@@ -40,13 +40,19 @@ pub struct AgentInfo {
...
     pub agent_latest_pubkey: AgentPubKey,
+    pub chain_head: (HeaderHash, u32, Timestamp),
 }
 
 impl AgentInfo {
-    pub fn new(agent_initial_pubkey: AgentPubKey, agent_latest_pubkey: AgentPubKey) -> Self {
+    pub fn new(
+        agent_initial_pubkey: AgentPubKey,
+        agent_latest_pubkey: AgentPubKey,
+        chain_head: (HeaderHash, u32, Timestamp),
```
Comment (artbrock) [inline crates/holochain_zome_types/src/info.rs:50]: I see you added a timestamp to the chain head if we have the actual timestamp that will be committed to the chain db while it's in the scratch space. If it is subject to change, I'm a little concerned about exposing it at this level.

I know we have system validation that ensures each chain entry is later than the prior entry, and I think it runs at the end of all the wasm processing when validating commits just before returning the result. So I'm a little concerned about this timestamp changing. But I guess if any timestamp changes, it changes the header hash which would change all following header hashes, so the hash is no more permanent than this timestamp anyway.

So... yeah... Maybe the timestamp will be useful to devs...

**Review by artbrock — APPROVED** [review]
Matches the requirements.

File: crates/holochain/src/core/ribosome/host_fn/call.rs [hunk 2]
```diff
@@ -122,11 +122,12 @@ pub mod wasm_test {
...
-                    AgentInfo {
-                        agent_initial_pubkey: bob_agent_id.clone(),
-                        agent_latest_pubkey: bob_agent_id.clone(),
-                    },
+                    &agent_info.agent_initial_pubkey,
+                    bob_agent_id
+                );
+                assert_eq!(
+                    &agent_info.agent_latest_pubkey,
+                    bob_agent_id
```
Comment (neonphog) [inline crates/holochain/src/core/ribosome/host_fn/call.rs:130]: just to have a test looking at the chain head here, is the sequence number consistent, could you check for that?

**Review by neonphog — APPROVED** [review]
looks good, potential suggestion for an additional test.

**Pull request merged**